src/components/node/NodeControls.tsx

```diff
@@ -25,6 +27,7 @@ const NodeControls: React.FC<NodeControlsProps> = ({ selectedNode, handleNodeSta
   const [input, setInput] = useState<string>('');
   const [isRunning, setIsRunning] = useState<boolean>(selectedNode.is_running);
   const outputRef = useRef<HTMLPreElement>(null);
+  const [messagePopup, setMessagePopup] = useState({ isOpen: false, message: '', title: '', type: 'info' as const });
```

Comment:
extract to type useState<PopupProps> - or something like this. Use it other components if they are using the same messagePopup structure.
Don't understand why "info as const" ?